onnxruntime/python/tools/transformers/models/phi2/convert_to_onnx.py

```diff
@@ -0,0 +1,544 @@
+        action="store_true",
+        help="Generate fp32 ONNX model for Nvidia GPUs",
+    )
+
+    parser.add_argument(
+        "--fp16_gpu",
+        required=False,
+        action="store_true",
+        help="Generate fp16 ONNX model for Nvidia GPUs",
+    )
+
+    parser.add_argument(
+        "--int4_gpu",
+        required=False,
+        action="store_true",
+        help="Generate int4 ONNX model for Nvidia GPUs",
+    )
+
+    parser.add_argument(
+        "--fp16_a100",
+        required=False,
+        action="store_true",
+        help="Generate fp16 ONNX model for Nvidia A100",
+    )
+
+    parser.add_argument(
+        "--int4_a100",
+        required=False,
+        action="store_true",
+        help="Generate int4 ONNX model for Nvidia A100",
+    )
+
+    parser.add_argument(
+        "--overwrite",
+        required=False,
+        action="store_true",
+        help="Overwrite existing ONNX models",
+    )
+
+    parser.add_argument(
+        "--cache_dir",
+        required=False,
+        type=str,
+        default="./cache",
+        help="The cache directory for the pytorch model",
+    )
+
+    parser.add_argument(
+        "--device_id",
+        required=False,
+        type=int,
+        default=0,
+        help="The device id for the pytorch model",
+    )
+
+    parser.add_argument(
+        "--run_example",
+        required=False,
+        action="store_true",
+        help="Run ORT inference example",
+    )
+
+    parser.add_argument(
+        "--skip_export",
+        required=False,
+        action="store_true",
+        help="Skip exporting ONNX model",
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = parse_arguments()
+
+    device = torch.device("cuda", args.device_id) if torch.cuda.is_available() else torch.device("cpu")
+
+    converter = ConvertPhi2ToONNX(device, cache_dir=args.cache_dir)
+
+    temp_onnx_path = "phi2_temp.onnx"
+    original_onnx_path = "phi2.onnx"
+
+    if not os.path.exists(original_onnx_path) or args.overwrite:
+        converter.dynamo_export(temp_onnx_path)
+        converter.preprocess_onnx(
+            temp_onnx_path,
+            original_onnx_path,
+            func_name="modeling_phi_PhiModel_model_1",  # The function to unroll
+        )
+        converter.erase_onnx_model(temp_onnx_path)
+
+    model_type_to_args = {
+        "fp32_cpu": (AttentionOpType.MultiHeadAttention, Precision.FLOAT32, "phi2_decoder_fp32_cpu.onnx"),
+        "int4_cpu": (AttentionOpType.MultiHeadAttention, Precision.INT4, "phi2_decoder_int4_cpu.onnx"),
+        "fp32_gpu": (AttentionOpType.Attention, Precision.FLOAT32, "phi2_decoder_fp32_gpu.onnx"),
+        "fp16_gpu": (AttentionOpType.Attention, Precision.FLOAT16, "phi2_decoder_fp16_gpu.onnx"),
+        "int4_gpu": (AttentionOpType.Attention, Precision.INT4, "phi2_decoder_int4_gpu.onnx"),
+        "fp16_a100": (AttentionOpType.GroupQueryAttention, Precision.FLOAT16, "phi2_decoder_fp16_a100.onnx"),
```

Comment:
Add a comment that this setting is for CUDA architecture SM=80~89 (like RTX 4090), not just for A100.